compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -2804,7 +2804,8 @@ public void visit(BLangTypeInit cIExpr, AnalyzerData data) {
         analyzeExprs(cIExpr.argsExpr, data);
         analyzeExpr(cIExpr.initInvocation, data);
         BType type = cIExpr.getBType();
-        if (cIExpr.userDefinedType != null && Symbols.isFlagOn(type.tsymbol.flags, Flags.DEPRECATED)) {
+        if (cIExpr.userDefinedType != null && type.tsymbol != null &&
+                Symbols.isFlagOn(type.tsymbol.flags, Flags.DEPRECATED)) {
```

Comment:
Here, we need to check whether the `DEPRECATED` flag is present in the symbol of the `userDefinedType` (`BLangUserDefinedType`), isn't it?